scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,279 @@
+const fs = require('fs');
+const { JSONPath } = require('jsonpath-plus');
+const { exec } = require('child_process');
+const path = require('path');
+const { Parser } = require('@asyncapi/parser');
+const parser = new Parser();
+
+// Read the markdown file
+const markdownContent = fs.readFileSync('../../spec/asyncapi.md', 'utf8');
+
+// Function to extract comments with example metadata
+function extractComments(content) {
+  const commentRegex = /<!-- asyncapi-example-tester:(\{.*?\}) -->/g;
+  let match;
+  const comments = [];
+
+  while ((match = commentRegex.exec(content)) !== null) {
+    try {
+      comments.push({
+        json: JSON.parse(match[1]),
+        index: match.index
+      });
+    } catch (e) {
+      console.error("Failed to parse comment JSON:", match[1], e);
+    }
+  }
+
+  return comments;
+}
+
+// Function to extract JSON examples from markdown content
+function extractExamples(content) {
+  const exampleRegex = /```json\s+([\s\S]*?)\s+```/g;
```

Comment:
Wouldn't make sense to create one single regex that gives you the object inside the comment/annotation plus the example?

Reply:
You can use the following regex that will give you 3 groups per match:

1. The raw JSON Object inside the comment, I.e. `{"name":"foo","json_path":"bar"}`
2. The format of the example (`json` | `yaml`)
3. The content of the example

Reply:
I'll try that!

Reply:
resolved through d7eb944